### PR TITLE
Evil Dataset Proof of Concept (updated proposal)

### PR DESCRIPTION
**Pull request by mikaylathompson**
### Description
This PR is a proof of concept for the updated proposal for the evil dataset (#9). It adds a single, fairly trivial example, along with the documentation/code samples to run it.

### Testing
Testing has been manual so far, and this is just POC for the time being, so I think that's sufficient until we've settled on a path forward.

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: evil-dataset/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+This directory is intended to house the "evil dataset"--a collection of data & associated queries that test edge cases and behavior changes between versions.
+
+## Datapoint Library
+The datapoints are contained in `/datapoint-library`. The directory name is the unique identifier for the test case.
+
+Within the datapoint-library directory, the directory structure should be as follows:
+
+```
+datapoint-library/
+├─ example-datapoint/
```
Comment (chelma) [inline evil-dataset/README.md:12]: [blocker] Happy to iterate on the design as appropriate, but I'm curious to hear your thoughts on the number of "datapoints" you think we'll have in the medium and long term and whether we'll want to stick with this approach of separate directory structures for each of them?  Or are we thinking of making a custom file format that encompasses all the data for a "datapoint"?  Or maybe multiple "datapoints"?

File: evil-dataset/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    "max_score" : 1.0,
+    "hits" : [ ... ]
+  }
+}
+
+# For ease of comparison, a jq filter can be provided, and this allows for a one-line curl command to compare the actual vs expected output. Any output from this command indicates a mismatch, silence means the query is as expected.
+> curl -s -XGET 'https://localhost:9200/_search?pretty' -ku "admin:admin" -H "Content-Type: application/x-ndjson" --data-binary @query.json | jq -f filter.jq | diff - expected.json
+
+# An unsuccesful comparison might look like the following:
+> curl -s -XGET 'https://localhost:9200/_search?pretty' -ku "admin:admin" -H "Content-Type: application/x-ndjson" --data-binary @query.json | jq -f filter.jq | diff - expected.json
```
Comment (chelma) [inline evil-dataset/README.md:69]: [Blocker] Trying to understand your longer-term intent here and what the scripting implications will be.  Do you see us using curl directly for the foreseeable future?  Or are we going to use the client SDKs for whatever language the test script is written in?

File: evil-dataset/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# Evil Dataset
+
+This directory is intended to house the "evil dataset"--a collection of data & associated queries that test edge cases and behavior changes between versions.
```
Comment (chelma) [inline evil-dataset/README.md:3]: [Blocker] Just trying to understand your intent.  Are you envisioning this housing all of our "expectations"?

**Review by chelma — COMMENTED** [review]
Hey there!  For context, when I leave comments in a PR I typically tag them as [blocking] or [non-blocking].

[blocking] comments are things I personally believe are important enough that I want to see them discussed/resolved before approving the change.

[non-blocking] comments are things I think would be nice to discuss/resolve, but not the end of the world if they are left as-is.  

Obviously, feel free to disregard if you aren't seeking my specific signoff! :-)
